internal/controller/replication.storage/status.go

```diff
@@ -76,24 +83,28 @@ func setFailedPromotionCondition(conditions *[]metav1.Condition, observedGenerat
 // sets conditions when volume promotion was failed due to failed validation.
 func setFailedValidationCondition(conditions *[]metav1.Condition, observedGeneration int64) {
 	setStatusCondition(conditions, &metav1.Condition{
+		Message:            "failed to promote",
 		Type:               v1alpha1.ConditionCompleted,
 		Reason:             v1alpha1.FailedToPromote,
 		ObservedGeneration: observedGeneration,
 		Status:             metav1.ConditionFalse,
 	})
 	setStatusCondition(conditions, &metav1.Condition{
+		Message:            "error detected while promotion",
```

Comment:
This condition is not related to promotion. Why do we need to talk about promotion here?

Reply:
this condition is reached while we were trying to promote and hence the error was detected right?